### ISSUE TITLE
Segfault when recalling a postponed message

### ISSUE THREAD

**Issue by xHire** [neomutt/neomutt#3268]
## Expected Behaviour

When I start neomutt, press `m` to write an e-mail, neomutt asks me `Recall postponed message?`, I confirm `yes` and neomutt proceeds into composing it.

## Actual Behaviour

After confirming `yes` to `Recall postponed message?`, neomutt crashes with a segmentation fault.

## Steps to Reproduce

* run neomutt
* press `m` to write an e-mail
* when neomutt asks `Recall postponed message?`, confirm `yes`
* SIGSEGV

I tried to strip down my setup to some small example that still reproduces the crash.

neomuttrc:
```
set folder = "~/.mem/bugreport/mail"
set mbox_type = maildir
set postponed = "+drafts"
set record = "+sent"
```

## How often does this happen?

- On a particular email. I suspect that the fact that it is a reply plays some role. A fresh postponed message in a fresh setup didn’t cause neomutt crash. This one does:

```
$ cat ~/.mem/bugreport/mail/drafts/cur/1645476461.R16767079780084815476.ryasu\:2\, 
Date: Mon, 21 Feb 2022 21:47:41 +0100
From: xhire@localhost
To: xHire <xhire@somewhere>
Subject: Re: foo
References: <1506100430.2755.126@somedomain.foo>
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Disposition: inline
Content-Transfer-Encoding: 8bit
In-Reply-To: <1506100430.2755.126@somedomain.foo>
X-Mutt-References: <1506100430.2755.126@somedomain.foo>
X-Mutt-Fcc: =sent

Oh, bug!
```

## When did it start to happen?

I’m a newcomer, so I was playing with configuration, learning neomutt interface, controls and behaviour, so I happened to randomly create this situation.

## NeoMutt Version

```
NeoMutt 20211029
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.16.5-gentoo (x86_64)
ncurses: ncurses 6.2.20210619 (compiled with 6.2.20210619)
libidn: 1.38 (compiled with 1.38)
GPGME: 1.16.0
GnuTLS: 3.7.2

Configure options: --prefix=/usr --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --docdir=/usr/share/doc/neomutt-20211029 --with-sysroot=/ --libdir=/usr/lib64 CCACHE=none CC_FOR_BUILD=x86_64-pc-linux-gnu-gcc --full-doc --enable-nls --disable-notmuch --enable-autocrypt --enable-gpgme --disable-pgp --disable-smime --disable-bdb --disable-gdbm --disable-kyotocabinet --disable-qdbm --disable-tokyocabinet --disable-lz4 --enable-zlib --enable-zstd --enable-idn --disable-gss --disable-lmdb --disable-sasl --with-ui=ncurses --sysconfdir=/etc/neomutt --enable-ssl --enable-gnutls --disable-testing

Compilation CFLAGS: -O0 -g -march=x86-64-v3 -pipe -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I//usr/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme -gss 
  -hcache -homespool +idn +inotify -locales_hack -lua -mixmaster +nls -notmuch 
  -openssl -pgp +regex -sasl -smime +sqlite +sun_attachment 

MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc/neomutt"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version: Gentoo Linux
* Were you using multiple copies of NeoMutt at once? No.
* Were you using 'screen' or 'tmux'? No.
* Is your email local (maildir) or remote (IMAP)? maildir

### Backtrace

I recompiled neomutt with `-O0 -g` and let it crash in gdb, this is `bt full`:

```
#0  0x00005555555ba046 in mutt_get_postponed (m_cur=0x0, hdr=0x555555a56fd0, cur=0x7fffffffcc68, fcc=0x7fffffffccc0) at postpone.c:721
        plen = 18
        c_postponed = 0x555555a22890 "/home/xhire/.mem/bugreport/mail/drafts"
        e = 0x555555a5fd60
        rc = 16
        p = 0x555555a55533 "<1506100430.2755.126@somedomain.foo>"
        m = 0x555555a57090
        __func__ = "mutt_get_postponed"
        c_delete = MUTT_ASKYES
        ctx = 0x555555a57ac0
        np = 0x555555a57cc0
        tmp = 0x555555a598a0
        c_crypt_opportunistic_encrypt = false
#1  0x0000555555669287 in mutt_send_message (flags=16, e_templ=0x555555a56fd0, tempfile=0x0, m=0x0, el=0x0, sub=0x5555557402f0) at send/send.c:2190
        buf = "(\210\323\367\377\177\000\000\310\020\325\367\377\177\000\000\000\000\000\000\000\000\000\000\n\000\000\000^\000\000\000\001\000\000\000\000\000\000\000\000]\321\367\377\177\000\000\001\000\000\000\377\177\000\000\360\362uUUU\000\001\320\315\377\377\377\177\000\000\326\315\377\377\377\177\000\000\371\001\000\000\000\000\000\000\377\001\000\000\000\000\000\000\033[348G\000\070H\000\377\377\377\177\000\000(\210\323\367\377\177\000\000\310\020\325\367\377\177\000\000\360\362uUUU\000\000\020\000\000\000^\000\000\000\001\000\000\000\000\000\000\000\000]\321\367\377\177\000\000\000\000\000\000\000\000\000\000\377\001", '\000' <repeats 22 times>, "\371\001\000\000\000\000\000\000"...
        fcc = {data = 0x555555a30d60 "", dptr = 0x555555a30d60 "", dsize = 1024}
        fp_tmp = 0x0
        pbody = 0x0
        i = 0
        free_clear_content = false
        clear_content = 0x0
        pgpkeylist = 0x0
        pgp_sign_as = 0x0
        smime_sign_as = 0x0
        tag = 0x0
        err = 0x0
        ctype = 0x0
        finalpath = 0x0
        en = 0x0
        e_cur = 0x0
        rc = -1
        c_recall = MUTT_ASKYES
        __func__ = "mutt_send_message"
        c_reverse_name = 255
        c_reply_with_xorig = 255
        c_resume_draft_files = 127
        c_copy = 32767
        c_abort_nosubject = 16777216
        c_fcc_before_send = false
#2  0x00005555555c7ca4 in op_mail (shared=0x555555a2ad00, priv=0x555555a23f70, op=132) at index/functions.c:702
        rc = 21845
#3  0x00005555555cd93f in index_function_dispatcher (win_index=0x555555a1b350, op=132) at index/functions.c:3021
        fn = 0x555555703c48 <IndexFunctions+1032>
        i = 43
        __func__ = "index_function_dispatcher"
        priv = 0x555555a23f70
        dlg = 0x555555757c50
        shared = 0x555555a2ad00
        rc = -7
#4  0x00005555555c5e97 in mutt_index_menu (dlg=0x555555757c50, m_init=0x0) at index/dlg_index.c:1341
        rc = 32767
        ctx_old = 0x0
        shared = 0x555555a2ad00
        panel_index = 0x555555757a20
        panel_pager = 0x555555a23870
        priv = 0x555555a23f70
        op = 132
        __func__ = "mutt_index_menu"
#5  0x00005555555a7787 in main (argc=1, argv=0x7fffffffdeb8, envp=0x7fffffffded8) at main.c:1377
        dlg = 0x555555757c50
        ev_m = {mailbox = 0x0}
        m = 0x0
        c_read_only = false
        subject = 0x0
        include_file = 0x0
        draft_file = 0x0
        new_type = 0x0
        dlevel = 0x0
        dfile = 0x0
        cli_nntp = 0x0
        e = 0x0
        attach = {stqh_first = 0x0, stqh_last = 0x7fffffffd830}
        commands = {stqh_first = 0x0, stqh_last = 0x7fffffffd840}
        queries = {stqh_first = 0x0, stqh_last = 0x7fffffffd850}
        alias_queries = {stqh_first = 0x0, stqh_last = 0x7fffffffd860}
        cc_list = {stqh_first = 0x0, stqh_last = 0x7fffffffd870}
        bcc_list = {stqh_first = 0x0, stqh_last = 0x7fffffffd880}
        sendflags = 0
        flags = 0 '\000'
        version = 0
        i = 70
        explicit_folder = false
        dump_variables = false
        one_liner = false
        hide_sensitive = false
        batch_mode = false
        edit_infile = false
        double_dash = 3
        nargc = 1
        rc = 1
        repeat_error = false
        folder = {data = 0x555555a2ce20 "/var/mail/xhire", dptr = 0x555555a2ce2f "", dsize = 128}
        expanded_infile = {data = 0x0, dptr = 0x0, dsize = 0}
        tempfile = {data = 0x0, dptr = 0x0, dsize = 0}
        cs = 0x55555573ed20
        __func__ = "main"
        rc2 = 0
        c_autocrypt = false
        c_folder = 0x555555a222f0 "/home/xhire/.mem/bugreport/mail"
```

If there’s anything else that could help you track this bug down, let me know.

**Comment by xHire** [comment]
Only after submitting, I realised I could try removing headers from the e-mail message and indeed, after removing `X-Mutt-References:` (while leaving everything else), neomutt doesn’t crash anymore. Perhaps also related that at the time of recalling, I’m not in the folder where the original message was.

**Comment by ukleinek** [comment]
I have the same issue, triggered by `neomutt -p`.

The problem is that in https://github.com/neomutt/neomutt/blob/1e6fbf9a1d1ada47e597efea4ef583becd587f29/postpone.c#L741 `mcur == NULL`.

**Comment by ukleinek** [comment]
Also happens on c12b986765091933bdadbf3d0a341e66bd42404a
